src/core/src/core_logic/PatchInstaller.py

```diff
@@ -132,6 +137,13 @@ def raise_if_telemetry_unsupported(self):
 
         self.composite_logger.log("{0}".format(Constants.TELEMETRY_COMPATIBLE_MSG))
 
+    def add_arch_dependencies_yum(self, package_manager, package, packages, package_versions, package_and_dependencies, package_and_dependency_versions):
+        package_name_without_arch = package_manager.get_product_name_without_arch(package)
+        for possible_arch_dependency, possible_arch_dependency_version in zip(packages, package_versions):
+            if package_manager.get_product_name_without_arch(possible_arch_dependency) == package_name_without_arch and possible_arch_dependency not in package_and_dependencies:
+                package_and_dependencies.append(possible_arch_dependency)
+                package_and_dependency_versions.append(possible_arch_dependency_version)
```

Comment:
Unless extenuating circumstances dictate, we must leave PatchInstaller generic and agnostic to individual package manager nuances. Please evaluate how this may be embedded within PackageManager generically and then specifically within YumPackageManager.

Reply:
Done